src/rust/ensogl/lib/theme/src/lib.rs

```diff
@@ -0,0 +1,229 @@
+    };
+
+    ([$($path:ident)*] $name:ident {$($t:tt)*}) => {
+        pub mod $name {
+            _define_theme_modules!([$($path)* $name] $($t)*);
+        }
+    };
+
+    ([$($path:ident)*] $name:ident {$($t:tt)*} $($rest:tt)*) => {
+        _define_theme_modules!([$($path)*] $name {$($t)*});
+        _define_theme_modules!([$($path)*] $($rest)*);
+    };
+}
+
+/// Used to define default theme.
+#[macro_export]
+macro_rules! define_default_theme {
+    ($name:ident $($t:tt)*) => {
+        define_theme!($name $($t)*);
+
+        #[allow(non_upper_case_globals)]
+        #[allow(missing_docs)]
+        pub mod vars {
+            _define_theme_modules!([$name] $($t)*);
+        }
+    };
+}
+
+/// Used to define any theme.
+#[macro_export]
+macro_rules! define_theme {
+    ($name:ident $($t:tt)*) => {
+        #[allow(missing_docs)]
+        pub mod $name {
+            use ensogl_core::application::Application;
+            use ensogl_core::data::color;
+            use ensogl_core::display::style::theme;
+
+            /// Setup the $name theme in application.
+            pub fn setup(app:&Application) {
+                let mut $name = theme::Theme::new();
+                _define_theme_literals!([$name] $($t)*);
+                app.themes.register(stringify!($name),$name);
+                app.themes.set_enabled(&[stringify!($name)]);
+            }
+        }
+    };
+}
+
+define_theme! { dark
+    application {
+        background {
+            color = color::Lcha::new(0.13,0.013,0.18,1.0)
+        }
+        text {
+            color = color::Lcha::new(1.0,0.0,0.0,0.7);
+            selection {
+                color = color::Lcha::new(0.7,0.0,0.125,0.7)
+            }
+        }
+    }
+    graph_editor {
+        node {
+            background {
+                color = color::Lcha::new(0.2,0.013,0.18,1.0)
+            }
+            selection {
+                color = color::Lcha::new(0.72,0.5,0.22,1.0);
+                size = 7.0
+            }
+        }
+        visualization {
+            background {
+                color = color::Lcha::new(0.2,0.013,0.18,1.0)
+            }
+        }
+    }
+    breadcrumbs {
+        full {
+            color = color::Lcha::new(1.0,0.0,0.0,0.7)
+        }
+        transparent {
+            color = color::Lcha::new(1.0,0.0,0.0,0.4)
+        }
+        selected {
+            color = color::Lcha::new(1.0,0.0,0.0,0.6)
+        }
+        deselected{
+            left {
+                color = color::Lcha::new(1.0,0.0,0.0,0.6)
+            }
+            right {
+                color = color::Lcha::new(1.0,0.0,0.0,0.2)
+            }
+        }
+        hover {
+            color = color::Lcha::new(1.0,0.0,0.0,0.6)
+        }
+    }
+    list_view {
+        background {
+            color = color::Lcha::new(0.2,0.013,0.18,1.0)
+        }
+        highlight {
+            color = color::Lcha::new(0.72,0.5,0.22,1.0)
+        }
+    }
```

Comment:
`list_view.{...}` -> `widget.list_view.{...}` there will be more widgets in future.